### PR TITLE
Integrate Vuex, Fetch Products from API, Wire Up Store and Product Pages

### PR DESCRIPTION
**Pull request by kylecorbelli**

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            [slug]: {
+              name,
+              imageUrl: `https://dsc-assets.imgix.net/images/product-images/product-tile/${sku}.png?auto=format&w=600`,
+              price,
+              sku,
+              slug,
+            },
+          }
+        },
+        {}
```
Comment (kylecorbelli) [inline src/store.ts]: comma dangle

File: src/components/StorePage.vue [hunk 2]
```diff
@@ -1,41 +1,78 @@
...
-        slug: 'shampoo',
-      },
-    ],
-  }),
+import {
+  mapActions,
+  mapGetters,
+} from 'vuex'
+
+interface StupidClass {
```
Comment (kylecorbelli) [inline src/components/StorePage.vue]: obviously need to type this better or figure out a cleaner solution. forcing this wacky interface just so typescript recognizes methods created by `mapActions` seems archaic

File: src/components/ProductPage.vue [hunk 3]
```diff
@@ -1,15 +1,52 @@
...
 </template>
 
 <script lang="ts">
-export default {}
+import {
+  mapActions,
+  mapGetters,
+} from 'vuex'
+
+interface ProductPage {
```
Comment (kylecorbelli) [inline src/components/ProductPage.vue]: obviously need to type this better or figure out a cleaner solution. forcing this wacky interface just so typescript recognizes methods created by `mapActions` seems archaic

Comment (briangonzalez) [reply src/components/ProductPage.vue]: Agreed. We'll figure it out. I am ok merging this one in.

**Pull request merged**